### DISCUSSION TITLE
Issues with using avatarName in filenames

### DISCUSSION THREAD

**Discussion by lyuma** [vrchat-community/osc#55] [General]
LocalAvatarData parameters are saved in:
```
C:\Users\USERNAME\AppData\LocalLow\VRChat\VRChat\LocalAvatarData\usr_xxxxxxxx-yyyy-zzzz-aaaa-bbbbccccdddd\
avtr_eeeeeeee-ffff-gggg-hhhh-iiiijjjjkkkk
```

In contrast, the new json files generated from OSC have the avatar's name in the filename, for example:
```
C:\Users\USERNAME\AppData\LocalLow\VRChat\VRChat\OSC\usr_xxxxxxxx-yyyy-zzzz-aaaa-bbbbccccdddd\Avatars\
my cool avatar with spaces_avtr_eeeeeeee-ffff-gggg-hhhh-iiiijjjjkkkk.json
```

I understand the idea was to make it easier for users to find it, but I have a few concerns with using the avatar's name in the filename:

- The avatar's name is difficult to know programmatically without doing an API request and knowing all the escaping rules (if any characters are illegal).
- What happens if I upload a new version of the avatar and change the name?
- Public avatars may have offensive names, that could leave unusually named files on my hard drive.

I would guess only technical users will know where to look in the first place, so I don't see the real benefit, especially given that you can browse https://vrchat.com/home and look up the avatar ID for your current avatar (or any of your own uploads).

Are there any other opinions on this? Just wanted to bring it up while there's still a chance for it to be changed. (also if anyone has feedback about my Av3Emulator and if it should incorporate support for reading this json data or not.)

**Comment by bdunderscore** [comment]
I agree with these points. While I find having access to this data useful, it's not really useful until I can automatically figure out which is the correct file. Even having the user browse to the file is problematic as they may not know their user ID, may have non-unique avatar names, or with public avatars might not know the public avatar name when they initially wear the avatar.

Since OSC needs to use an external tool in any case, it might be best to name things in a way easily found by a tool and to provide the information needed to find the path over OSC (in vein with #5)

I'll note however that whatever files are created in the v1 will become an API, of sorts, that applications and tools depend on, so if this naming scheme is going to change I'd suggest doing so ASAP, prior to the full release.
